antarest/study/storage/rawstudy/model/filesystem/inode.py

```diff
@@ -124,6 +124,16 @@ def denormalize(self) -> None:
         """
         raise NotImplementedError()
 
+    def get_relative_path_inside_archive(self) -> Path:
+        archive_path = self.config.archive_path
+        if archive_path:
+            return self.config.path.relative_to(archive_path.parent / self.config.study_id)
+        else:
+            raise ShouldNotHappenException(
```

Comment:
You should raise another Exception as this one is really for cases that shouldn't happen. Here it can happen if someone misuses the method.